### ISSUE TITLE
Analyzer version conflict with json_serializable package

### ISSUE THREAD

**Issue by lkloon123** [numen31337/copy_with_extension#2]
json_serializable require analyzer version `analyzer: '>=0.37.1 <0.39.0'`
this package require analyzer version `analyzer: ">=0.39.0 <1.0.0"`

Therefore the pubspec is unable to be resolved.

Temporary solution
```yaml
# add this line to pubspec.yaml
dependency_overrides:
  analyzer: '>=0.37.1 <0.39.0'
```

This should be able to resolve when the time `json_serializable` update their package

**Comment by numen31337** [comment]
I'm aware that `json_serializable` rely on `analyzer` `<0.39.0` as they use some deprecated features of `analyzer`. Unfortunately, in order to correctly generate the extensions, we need `analyzer` `>=0.39.0` as starting from this particular version it supports the extensions.

**Comment by lkloon123** [comment]
this is fixed for `json_serializable` `>=3.2.4`
